src/main/java/org/broadinstitute/hellbender/tools/spark/sv/AssemblyAlignmentParser.java

```diff
@@ -1,112 +1,35 @@
-     */
-    static JavaPairRDD<Iterable<AlignmentRegion>, byte[]> prepAlignmentRegionsForCalling(final JavaSparkContext ctx,
-                                                                                         final String pathToInputAlignments,
-                                                                                         final String pathToInputAssemblies) {
-
-        final JavaPairRDD<Tuple2<String, String>, Iterable<AlignmentRegion>> alignmentRegionsKeyedByAssemblyAndContigId
-                = parseAlignments(ctx, pathToInputAlignments);
-        if (DEBUG_STATS) {
-            AssemblyAlignmentParser.debugStats(alignmentRegionsKeyedByAssemblyAndContigId, pathToInputAlignments);
-        }
-
-        final JavaPairRDD<Tuple2<String, String>, byte[]> contigSequences
-                = loadContigSequence(ctx, pathToInputAssemblies);
-
-        return alignmentRegionsKeyedByAssemblyAndContigId
-                .join(contigSequences)
-                .mapToPair(Tuple2::_2);
-    }
-
-    // TODO: 11/23/16 test
-    static JavaPairRDD<Tuple2<String, String>, Iterable<AlignmentRegion>> parseAlignments(final JavaSparkContext ctx,
-                                                                                          final String pathToInputAlignments) {
-
-        return ctx.textFile(pathToInputAlignments).map(textLine -> AlignmentRegion.fromString(textLine.split(AlignmentRegion.STRING_REP_SEPARATOR,-1)))
-                .flatMap(oneRegion -> breakGappedAlignment(oneRegion, SVConstants.DiscoveryStepConstants.GAPPED_ALIGNMENT_BREAK_DEFAULT_SENSITIVITY).iterator())
-                .mapToPair(alignmentRegion -> new Tuple2<>(new Tuple2<>(alignmentRegion.assemblyId, alignmentRegion.contigId), alignmentRegion))
-                .groupByKey();
-    }
-
-    // TODO: 12/15/16 test
-    static JavaPairRDD<Tuple2<String, String>, byte[]> loadContigSequence(final JavaSparkContext ctx,
-                                                                          final String pathToInputAssemblies) {
-        return loadContigsCollectionKeyedByAssemblyId(ctx, pathToInputAssemblies)
-                .flatMapToPair(assemblyIdAndContigsCollection -> {
-                    final String assemblyId = assemblyIdAndContigsCollection._1;
-                    final ContigsCollection contigsCollection = assemblyIdAndContigsCollection._2;
-                    return contigsCollection.getContents().stream()
-                            .map(pair -> new Tuple2<>(new Tuple2<>(assemblyId, pair._1.toString()), pair._2.toString().getBytes()))
-                            .collect(Collectors.toList()).iterator();
-                });
-    }
-
-    // TODO: 1/18/17 handle mapping quality and mismatches of the new alignment better rather than artificial ones. this would ultimately require a re-alignment done right.
+    public JavaRDD<AlignedAssembly.AlignedContig> loadAndFilterAndParseContigAlignments(final JavaSparkContext ctx,
```

Comment:
After seeking suggestion from @tedsharpe , I've replaced the `Interface` with an abstract class `AlignedContigGenerator`. The intention was really to signify an adaptation of various evidence-finding stage formats (SAM, BwaMemAilgnment, SGA-route-generated text records) to a single discovery stage format.